### DISCUSSION TITLE
[Feature request] Improve documentation

### DISCUSSION THREAD

**Discussion by bunny1985** [ing-bank/lion#1523] [General] [answered]
The documentation at current state is really limited. 
For example **components events seems to be not documented at all.** 
There is no "bootstrap application" example. 
Form programming paradigm  is explained poorly.  There should be at least couple of examples with "on change" validation "on submit" validation and how to properly grab form all form values. 
Ajax's interceptors are only mentioned and never explained. 
Pls. do something about it. In library projects like this, documentation is equally important as code itself. or even more important.

**Comment by daKmoR** [comment, marked as answer]
I and I'm pretty sure the whole team fully agrees with that 🤗

that being said our resources are pretty limited currently 🙈  business requires is to work on some internal things based on lion... 

however, if you want to help out by starting some docs for things you think are important we can surely review and help fill the gaps. If that is too much work raising issues with specific topics and questions could help us prioritize those with our limited time.
An example could be

- Add explanation for interceptors
  - Show how to add an interceptor
  - Show how to do "specific usecase"
  - Show hot to do "specific user case 2"

- Add example for onsubmit on "link to page"
  - Should have at least ...
  - Needs to show that this and that works...

**Comment by bunny1985** [comment]
thanks for honest response. 
I only have hope that it will change since currently working with ing-lion is difficult and entire project looses in eyes of developers.
